Description: fix ntp-keygen infinite loop or lack of randonmess on big
 endian platforms
Origin: backport, http://bk1.ntp.org/ntp-stable/?PAGE=patch&REV=55199296N2gFqH1Hm5GOnhrk9Ypygg
Bug: https://bugs.ntp.org/show_bug.cgi?id=2797

Index: ntp-4.2.6.p5+dfsg/util/ntp-keygen.c
===================================================================
--- ntp-4.2.6.p5+dfsg.orig/util/ntp-keygen.c	2015-04-13 08:58:49.317778349 -0400
+++ ntp-4.2.6.p5+dfsg/util/ntp-keygen.c	2015-04-13 08:58:49.317778349 -0400
@@ -726,24 +726,24 @@
 	ntp_srandom((u_long)epoch);
 	for (i = 1; i <= MD5KEYS; i++) {
 		for (j = 0; j < MD5SIZE; j++) {
-			int temp;
+			u_char temp;
 
 			while (1) {
 				int rc;
 
-				rc = ntp_crypto_random_buf(&temp, 1);
+				rc = ntp_crypto_random_buf(
+				    &temp, sizeof(temp));
 				if (-1 == rc) {
 					fprintf(stderr, "ntp_crypto_random_buf() failed.\n");
 					exit (-1);
 				}
-				temp &= 0xff;
 				if (temp == '#')
 					continue;
 
 				if (temp > 0x20 && temp < 0x7f)
 					break;
 			}
-			md5key[j] = (u_char)temp;
+			md5key[j] = temp;
 		}
 		md5key[j] = '\0';
 		fprintf(str, "%2d MD5 %s  # MD5 key\n", i,
